Source/UnitTests/ExecutionEngineTests/ExecutionEngineTest.cs

```diff
@@ -329,26 +359,24 @@ public async Task LoopInvariantDescriptions()
     Assert.AreEqual((firstName, new Running()), statusList[0]);
     Assert.AreEqual((secondName, new Queued()), statusList[1]);
     Assert.AreEqual(firstName, statusList[2].Item1);
-    Assert.IsTrue(statusList[2].Item2 is BatchCompleted);
-    Assert.AreEqual(firstName, statusList[3].Item1);
-    Assert.IsTrue(statusList[3].Item2 is Completed);
-    Assert.AreEqual((secondName, new Running()), statusList[4]);
-    Assert.AreEqual(secondName, statusList[5].Item1);
-    Assert.IsTrue(statusList[5].Item2 is BatchCompleted);
-    Assert.AreEqual(secondName, statusList[6].Item1);
-    Assert.IsTrue(statusList[6].Item2 is Completed);
-    
-    var tasks2 = engine.GetImplementationTasks(program);
-    Assert.True(tasks2[0].CacheStatus is Completed);
-    Assert.AreEqual(VcOutcome.Errors, ((Completed)tasks2[0].CacheStatus).Result.VcOutcome);
-
-    Assert.True(tasks2[1].CacheStatus is Completed);
-    Assert.AreEqual(VcOutcome.Correct, ((Completed)tasks2[1].CacheStatus).Result.VcOutcome);
+    Assert.AreEqual(firstName, statusList[2].Item1);
+    Assert.IsTrue(statusList[2].Item2 is Completed);
+    Assert.AreEqual((secondName, new Running()), statusList[3]);
+    Assert.AreEqual(secondName, statusList[4].Item1);
+    Assert.IsTrue(statusList[4].Item2 is Completed);
+  
+    // Caching is currently not working
```

Comment:
Given that at least this test of caching was working before, it'd be unfortunate to merge something that breaks it. Since I know that's something you're working on, do you have an upcoming PR already planned that'll fix it? It's perhaps worth making that separate for ease of reviewing, but I'd be reluctant to push a release of Boogie until it's at least working as well as it was before.

Reply:
Caching is currently not used by Dafny, because we don't trust it. This pathway of Boogie, for which caching does not work, is only used by Dafny. To 'fix' the caching, I would not to write some new code that does caching at the split level instead of the implementation level. Probably nothing complex, but it does not make sense to me to develop something that we won't test end-to-test because we're not using it.

Reply:
If you're confident that it's not used by any other tools, then this sounds fine.